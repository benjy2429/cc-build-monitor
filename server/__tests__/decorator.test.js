import decorateProject from '../decorator';

describe('Decorator', () => {
  describe('Projects', () => {
    const input = {
      _name: 'name',
      _activity: 'activity',
      _lastBuildStatus: 'lastBuildStatus',
      _lastBuildLabel: 'lastBuildLabel',
      _lastBuildTime: '2000-01-01',
    };

    it('decorates the input', () => {
      expect(decorateProject(input)).toMatchSnapshot();
    });

    it('ignores extra keys', () => {
      expect(decorateProject({ ...input, extra: 'extra' })).toMatchSnapshot();
    });

    it('defaults to undefined', () => {
      expect(decorateProject({})).toMatchSnapshot();
    });
  });
});
