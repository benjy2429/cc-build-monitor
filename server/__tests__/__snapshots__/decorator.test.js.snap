// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decorator Projects decorates the input 1`] = `
Object {
  "activity": "activity",
  "buildNumber": "lastBuildLabel",
  "buildStatus": "lastBuildStatus",
  "buildTime": 2000-01-01T00:00:00.000Z,
  "name": "name",
}
`;

exports[`Decorator Projects defaults to undefined 1`] = `
Object {
  "activity": undefined,
  "buildNumber": undefined,
  "buildStatus": undefined,
  "buildTime": undefined,
  "name": undefined,
}
`;

exports[`Decorator Projects ignores extra keys 1`] = `
Object {
  "activity": "activity",
  "buildNumber": "lastBuildLabel",
  "buildStatus": "lastBuildStatus",
  "buildTime": 2000-01-01T00:00:00.000Z,
  "name": "name",
}
`;
